src/custom-elements/src/components/popover/popover.tsx

```diff
@@ -394,62 +396,8 @@ export class Popover {
     if (this.drag) {
       const popoverHeader = this._popoverElement
         .querySelector(`.${POPOVER_CLASSES.HEADER}`) as HTMLElement;
-      let initialTransformProperty: string;
-      const dragMouseDown = (e: MouseEvent) => {
-        e.preventDefault();
-        this._closePrevented = true;
-        pos3 = e.clientX;
-        pos4 = e.clientY;
-        initialTransformProperty = window.getComputedStyle(this._popoverElement)['transform'];
-
-        document.onmouseup = closeDragElement;
-        document.onmousemove = elementDrag;
-      };
-      const dragTouch = (e: TouchEvent) => {
-        e.preventDefault();
-        this._closePrevented = true;
-        pos3 = e.touches[0].clientX;
-        pos4 = e.touches[0].clientY;
-        initialTransformProperty = window.getComputedStyle(this._popoverElement)['transform'];
-
-        popoverHeader.ontouchmove = elementDragOnTouch;
-      };
-      const setPosition = (clientX: number, clientY: number) => {
-        pos1 = pos3 - clientX;
-        pos2 = pos4 - clientY;
-        pos3 = clientX;
-        pos4 = clientY;
-        this._popoverElement.style.top = this._popoverElement.offsetTop - pos2 + "px";
-        this._popoverElement.style.left = this._popoverElement.offsetLeft - pos1 + "px";
-        this._popoverElement.style.transform = initialTransformProperty;
-      };
-      const elementDrag = (e: MouseEvent) => {
-        e.preventDefault();
-        this._destroyPopper();
-        setPosition(e.clientX, e.clientY);
-      };
-      const elementDragOnTouch = (e: TouchEvent) => {
-        e.preventDefault();
-        const touch = e.targetTouches[0];
 
-        this._destroyPopper();
-        setPosition(touch.clientX, touch.clientY);
-      };
-      const closeDragElement = () => {
-        document.onmouseup = null;
-        document.onmousemove = null;
-      };
-      let pos1 = 0,
-        pos2 = 0,
-        pos3 = 0,
-        pos4 = 0;
-
-      if (popoverHeader) {
-        popoverHeader.onmousedown = dragMouseDown;
-        popoverHeader.ontouchstart = dragTouch;
-      } else {
-        this._popoverElement.onmousedown = dragMouseDown;
-      }
+      new Drag(popoverHeader, this._popoverElement, this);
```

Comment:
Please, save the Drag object in a class property, so when the popover is disposed you can do a `this._drag.dispose()` as well.

Reply:
Also, the `this._drag = new Drag(popoverHeader, this._popoverElement, this);` should be in a different method, decorated with a watcher of the `drag` prop. Like this:
```TypeScript
@Watch("drag")
function dragUpdated(newVal: boolean) {
  if (newVal && !this._drag) {
    this._drag = new Drag(popoverHeader, this._popoverElement, this);
  } else if (!newVal && this._drag){
    this._drag.dispose();
    this._drag = undefined;
  }
}
```
As the watcher might not be thrown at the beginning, you'll need to call this function from the didLoad as well:
```Typescript
  componentDidLoad(): void {
    // ...
    this.dragUpdated(this.drag);
  }

  componentDidUnload(): void {
    // ...
    if (this._drag) {
      this._drag.dispose();
      this._drag = undefined;
    }
```